pytest/tests/mocknet/s3_state_sync.py

```diff
@@ -0,0 +1,159 @@
+                'max': 0
+            },
+        }
+
+    # if late is True, returns late validators
+    # if late is False, returns not late validators
+    # if late is None, returns all validators
+    def get_validators(self, late=None):
+        # TODO(posvyatokum) this will depend on terraform script implementation
+        labels = {'role': 'validator'}
+        if late is not None:
+            labels['late'] = 'yes' if late else 'no'
+        return get_nodes_by_label(self.nodes, {'role': 'validator'})
+
+    def get_dumping_nodes(self):
+        # TODO(posvyatokum) this will depend on terraform script implementation
+        return get_nodes_by_label(self.nodes, {'state-parts-producer': 'yes'})
+
+    def get_late_nodes(self):
+        # TODO(posvyatokum) this will depend on terraform script implementation
+        return get_nodes_by_label(self.nodes, {'late': 'yes'})
+
+    def setup_dump_nodes(self):
+        if self.localnet:
+            set_state_parts_dump_mode_for_nodes(self.get_dumping_nodes(),
+                                                root_dir=self.parts_local_dir)
+        else:
+            set_state_parts_dump_mode_for_nodes(self.get_dumping_nodes(),
+                                                bucket=self.s3_bucket,
+                                                region=self.s3_region)
+
+        for i, node in enumerate(self.get_dumping_nodes()[:2]):
+            set_tracked_shard_for_node(node, [i * 2, i * 2 + 1])
+
+    # Starts the traffic node in mirroring mode
+    def start_traffic(self):
+        # TODO(posvyatokum): implement
+        pass
+
+    def run(self):
+        all_stats = []
+
+        logger.info("Step 0: setup nodes")
+        stop_nodes(self.nodes)
+        init_nodes(self.nodes, self.traffic_nodes)
+        self.setup_dump_nodes()
+
+        logger.info("Step 1: run early nodes for 5 epochs")
+        early_nodes = exclude_nodes(self.nodes, self.get_late_nodes())
+        restart_nodes(early_nodes)
+        wait_for_nodes_to_be_up(early_nodes)
+        self.start_traffic()
+        wait_n_epochs(5,
+                      early_nodes,
+                      all_stats=all_stats,
+                      stats_bounds=self.stats_bounds)
+
+        all_late_nodes = self.get_late_nodes()
+
+        logger.info("Step 2: setup and run late validator for 2 epochs")
+        late_validators = self.get_validators(late=True)
+        set_tracked_shard_for_nodes(late_validators, [])
+        restart_nodes(late_validators)
+        wait_for_nodes_to_be_up(late_validators)
+        wait_n_epochs(2,
+                      early_nodes + late_validators,
+                      all_stats=all_stats,
+                      stats_bounds=self.stats_bounds)
+
+        logger.info(
+            "Step 3: setup and run late rpc; wait until late nodes sync")
+        late_rpc = exclude_nodes(self.get_late_nodes(), late_validators)
+        set_tracked_shard_for_nodes(late_validators, [1, 2, 3, 4])
```

Comment:
```suggestion
        set_tracked_shard_for_nodes(late_validators, [0])
```

Confusingly, `tracked_shards=[0]` and `tracked_shards=[123,234,345]` mean exactly the same thing, i.e. track all shards.